packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -63,7 +63,7 @@ function getNextPublicEnvironmentVariables(): DefineEnv {
   for (const key in process.env) {
     if (key.startsWith('NEXT_PUBLIC_')) {
       const value = process.env[key]
-      if (value) {
+      if (value != null) {
```

Comment:
```suggestion
      if (typeof value === 'string') {
```

Why don't we check if is string? It'd be `''` right?

Reply:
Yeah, it is a noop